crates/sui-core/src/consensus_validator.rs

```diff
@@ -214,20 +215,32 @@ impl SuiTxValidator {
     fn vote_transaction(
         &self,
         epoch_store: &Arc<AuthorityPerEpochStore>,
-        tx: TransactionWithAliases,
+        tx: PlainTransactionWithClaims,
     ) -> SuiResult<()> {
-        let (tx, aliases) = tx.into_inner();
+        // Extract claims before consuming the transaction
+        let aliases = tx.aliases().clone();
```

Comment:
What's the reason for using `NonEmpty` as return type? `aliases()` should not panic when validating arbitrary input.